src/GoTrueClient.ts

```diff
@@ -1198,6 +1198,11 @@ export default class GoTrueClient {
           throw error
         }
 
+        if (!data.session?.access_token) {
+          // if there's no access token, the user can't be fetched
+          return { data: { user: null }, error: null }
```

Comment:
hmm should we consider returning the `AuthSessionMissingError` instead of null? i feel like that makes the DX better because the developer can check against that and redirect the user to the login page

Reply:
I'd say returning null is best here, because `getUser` is just trying to read information; similar to `getSession`.

Currently, the `AuthSessionMissingError` is only thrown for methods which try to change or set something - update a user, refresh a session, set a session, reauthenticate; none of which will succeed unless there is a session. By contrast, `getUser` has no real success/fail characteristic - the user info is either there or it isn't.

Reply:
Actually, this can't return an error, or it's going to cause issues with https://github.com/supabase/auth-js/pull/874

Reply:
I think we'll return error, and deal with #874 differently...